xarray/plot/utils.py

```diff
@@ -221,8 +221,14 @@ def _infer_xy_labels(darray, x, y):
         if darray.ndim != 2:
             raise ValueError('DataArray must be 2d')
         y, x = darray.dims
-    elif x is None or y is None:
-        raise ValueError('cannot supply only one of x and y')
+    elif x is None:
+        if y not in darray.dims:
+            raise ValueError('y must be a coordinate variable')
```

Comment:
I'm not sure I understand the logic. The user requests a y variable that is not in the coordinates. It would not work even if she supplied a valid x coordinate. Your messages suggests it would.

Maybe "y is not a coordinate variable" is more to the point (and the same for x)?